packages/@aws-cdk/aws-codepipeline-actions/lib/ecr/source-action.ts

```diff
@@ -93,9 +94,14 @@ export class EcrSourceAction extends Action {
       resources: [this.props.repository.repositoryArn],
     }));
 
+    let imageTag: string | undefined;
+    if (this.props.imageTag !== '') {
+      imageTag = this.props.imageTag ?? 'latest';
+    }
+
     this.props.repository.onCloudTrailImagePushed(Names.nodeUniqueId(stage.pipeline.node) + 'SourceEventRule', {
       target: new targets.CodePipeline(stage.pipeline),
-      imageTag: this.props.imageTag ?? 'latest',
+      imageTag,
```

Comment:
Since we're leaving the below expression unchanged, I would get rid of this `imageTag` variable altogether:

```suggestion
      imageTag: this.props.imageTag === '' ? undefined : (this.props.imageTag ?? 'latest'),
```